phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewIT.java

```diff
@@ -388,51 +435,65 @@ public void testViewAndTableInDifferentSchemas(boolean isNamespaceMapped) throws
         } catch (TableNotFoundException ignore) {
         }
         ddl = "DROP TABLE " + fullTableName1;
-        validateCannotDropTableWithChildViewsWithoutCascade(conn, fullTableName1);
         ddl = "DROP VIEW " + fullViewName2;
         conn.createStatement().execute(ddl);
         ddl = "DROP TABLE " + fullTableName1;
         conn.createStatement().execute(ddl);
     }
 
-    
+
     @Test
-    public void testDisallowDropOfColumnOnParentTable() throws Exception {
+    public void testDropOfColumnOnParentTableInvalidatesView() throws Exception {
         Connection conn = DriverManager.getConnection(getUrl());
+        String fullTableName = generateUniqueTableName();
+        String viewName = generateUniqueViewName();
+        splitSystemCatalog(Lists.newArrayList(fullTableName, viewName));
+
         String ddl = "CREATE TABLE " + fullTableName + " (k1 INTEGER NOT NULL, k2 INTEGER NOT NULL, v1 DECIMAL, CONSTRAINT pk PRIMARY KEY (k1, k2))" + tableDDLOptions;
         conn.createStatement().execute(ddl);
-        String viewName = "V_" + generateUniqueName();
         ddl = "CREATE VIEW " + viewName + "(v2 VARCHAR, v3 VARCHAR) AS SELECT * FROM " + fullTableName + " WHERE v1 = 1.0";
         conn.createStatement().execute(ddl);
         
-        try {
-            conn.createStatement().execute("ALTER TABLE " + fullTableName + " DROP COLUMN v1");
-            fail();
-        } catch (SQLException e) {
-            assertEquals(SQLExceptionCode.CANNOT_MUTATE_TABLE.getErrorCode(), e.getErrorCode());
+        conn.createStatement().execute("ALTER TABLE " + fullTableName + " DROP COLUMN v1");
+        // TODO see if its possibel to prevent the dropping of a column thats required by a child view (for its view where clause)
+        // the view should be invalid
```

Comment:
I think it's fine to consider the view invalid (i.e. fail any query that attempts to use it) if all it's columns can no longer be found. This is pretty typical in RDBMS.

Reply:
Previously we used to prevent a base table column from being dropped if the column was used in a child view index. 
It's possible to detect this and prevent the column drop even with splittable system.catalog if you think we should maintain the existing behavior.

Reply:
Seems like there'd be an inherent race condition, but if not, I suppose keeping the same behavior is fine. What about preventing the drop of a table with child views?

Reply:
I was thinking of handling the race condition by writing a cell using checkAndPut with the column name being mutated (added or dropped) to the header row of the base table. When a view is create the columns used in the view where clause will also do a checkAndPut on this cell in the base table. If this is happening concurrently only of the clients will succeed (either the drop column, or the view creation). 

If a DROP TABLE cascade is issued then the base table (with child views) is dropped. When any child views are resolved and we see the base table does not exists we throw a TableNotFoundException. If the DROP VIEW is issued without CASCADE and there are child views the statement fails with a not allowed to mutate exception. If a drop table without cascade and a create view happens concurrently we could create the view even though the base table would have been dropped, but the next time the view is resolved you would get a TableNotFoundException.

Reply:
I think we should avoid needing any kind of locking (including a checkAndPut). The scaling issues we ran into were caused by contention on a lock for the parent. The same thing could happen again with a checkAndPut (which is just locking the row during the check). I'd err on the side of scalability and have a view be invalidated if its parent is deleted. I think that's a more scalable solution.

Reply:
@JamesRTaylor 

I think we can avoid this while dropping a column. However while adding a column as the view and base table could be on different region servers we need to detect and prevent a conflicting add column change happening at the same time. I couldn't think of a better way to prevent this other than doing a checkAndPut. 
If we do the checkAndPut in MetadataClient to ensure that only one client is able to add the same column at the same time and then make the rpc to the server to add the column, do you think think that would scale?

Reply:
@JamesRTaylor 

We don't need to do the checkAndPut on the header row of the base table as this will block clients making changes to different columns. We only need to ensure that if multiple clients are making changes to the same column only one of them is allowed to make the change. So we can use the rowkey of the column row. We could also do the checkAndPut on the new SYSTEM.CHILD_LINK table instead of SYSTEM.CATALOG. This would be done from MetadataClient to ensure that no other client can make a conflicting change before we call addColumn or dropColumn.

Reply:
Ok, this sounds like it'll work fine.